houston/ardu/command_factory.py

```diff
@@ -0,0 +1,95 @@
+import os
+import logging
+import yaml
+from typing import Dict, Any, Type, List
+
+from .connection import CommandLong
+from ..valueRange import ContinuousValueRange, DiscreteValueRange
+from ..command import Parameter, Command, CommandMeta
+from ..specification import Idle
+
+logger = logging.getLogger(__name__)  # type: logging.Logger
+logger.setLevel(logging.DEBUG)
+
+
+def create_command(command: Dict[str, Any]) -> Type[Command]:
+    """
+    From a given dictionary, generates the Command class.
+    """
+    try:
+        name = command['name']
+    except KeyError:
+        msg = "missing 'name' field of Command"
+        raise TypeError(msg)
+    try:
+        id = command['id']
+    except KeyError:
+        msg = "missing 'id' field of Command"
+        raise TypeError(msg)
+    parameters = []
```

Comment:
Do we need both `parameters` and `params_name`? Also, `name_to_param` might be a slightly more readable name than `params_name`. (A type annotation would also be great.)